### PR TITLE
Use Redux for managing app state

### PR DESCRIPTION
**Pull request by kfarnung**
Remove the current local storage approach in favor of Redux and Redux-Persist. This moves most of the state management logic out of the components and lets them be relatively simple interface elements.

One of the big drawbacks is that the actions need to process primitive types, but this was pretty straightforward to work around. This may inform future versions of puzzle-lib to make the API surface more conducive to immutable uses.

### CODE REVIEW

File: src/Features/Cipher/Autokey/autokeyCipherSlice.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  secret: '',
+  text: '',
+};
+
+const autokeyCipherSlice = createSlice({
+  name: 'autokeyCipher',
+  initialState,
+  reducers: {
+    clear(state) {
+      state.conversion = 2;
```
Comment (beckbria) [inline src/Features/Cipher/Autokey/autokeyCipherSlice.ts:20]: Could this be either

state = initialState
Or
state.converstion = initialState.converstion // etc.

File: src/Features/Cipher/KeyedCipherStream/index.tsx [hunk 2]
```diff
@@ -8,68 +8,41 @@ import FormControl from 'react-bootstrap/FormControl';
...
   function calculateOutput() {
-    props.cipher.key = key;
-    props.cipher.text = text;
-    return conversion === 1 ?
-      props.cipher.encrypt() :
-      props.cipher.decrypt();
+    const cipher = props.cipher;
+    cipher.key = props.secret;
+    cipher.text = props.text;
+    return props.conversion === 1 ? cipher.encrypt() : cipher.decrypt();
```
Comment (beckbria) [inline src/Features/Cipher/KeyedCipherStream/index.tsx:45]: Not new in your change, but this feels like a magic number to me - should we have a constant for this?  Should it just be a boolean?

**Comment by kfarnung** [comment]
Live demo: https://kfarnung.github.io/puzztool/

File: src/Features/Cipher/Autokey/AutokeyStream.tsx [hunk 3]
```diff
@@ -1,19 +1,43 @@
...
   prompt: JSX.Element | string;
 }
 
 function AutokeyStream(props: Props) {
   return (
     <KeyedCipherStream
       cipher={new AutoKeyString()}
+      conversion={props.conversion}
       id="AutokeyStream"
+      secret={props.secret}
```
Comment (kfarnung) [inline src/Features/Cipher/Autokey/AutokeyStream.tsx:32]: Fun fact: I originally called this property `key` without realizing that `key` is a special property in React. Took quite a while to debug that issue...

**Pull request merged**